manticore/seth.py

```diff
@@ -885,188 +977,253 @@ def global_findings(self):
-                            state.constrain(memlog== res)
+    def _generate_testcase_callback(self, state, name, message):
+        '''
+        Create a serialized description of a given state.
+        :param state: The state to generate information about
+        :param message: Accompanying message
+        '''
+        # workspace should not be responsible for formating the output
+        # each object knows its secrets, each class should be able to report its
+        # final state
+        #super(ManticoreEVM, self)._generate_testcase_callback(state, name, message)
+        def flagged(flag):
+            return '(*)' if flag else '' 
+        testcase = self._output.testcase()
+        logger.info("Generated testcase No. {} - {}".format(testcase.num, message))
+        blockchain = state.platform
+        with testcase.open_stream('summary') as summary:            
+            summary.write("Last exception: %s\n" %state.context['last_exception'])
+
+            #Last instruction
+            metadata = self.get_metadata(blockchain.transactions[-1].address)
+            if metadata is not None:
+                address, offset = state.context['seth.trace'][-1]
+                summary.write('Last instruction at contract %x offset %x\n' %(address, offset))
+                at_runtime = blockchain.transactions[-1].sort != 'Create'
+                summary.write(metadata.get_source_for(offset, at_runtime))
+                summary.write('\n')
+
+
+
+            #Accounts summary
+            is_something_symbolic = False
+            summary.write("%d accounts.\n" % len(blockchain.accounts))
+            for account_address in blockchain.accounts:
+                summary.write("Address: 0x%x\n" % account_address)
+                balance = blockchain.get_balance(account_address)
+                is_balance_symbolic = issymbolic(balance)
+                is_something_symbolic = is_something_symbolic or is_balance_symbolic
+                balance = state.solve_one(balance)
+                summary.write("Balance: %d %s\n" % (balance, flagged(is_balance_symbolic)))
+
+                if blockchain.has_storage(account_address):
+                    summary.write("Storage:\n")
+                    for offset, value in blockchain.get_storage_items(account_address):
+                        is_storage_symbolic = issymbolic(offset) or issymbolic(value) 
+                        offset = state.solve_one(offset)
+                        value = state.solve_one(value)
+                        summary.write("\t%032x -> %032x %s\n" % (offset, value, flagged(is_storage_symbolic)))
+                        is_something_symbolic = is_something_symbolic or is_storage_symbolic
+
+                code = blockchain.get_code(account_address)
+                if len(code):
+                    summary.write("Code:\n")
+                    fcode = StringIO.StringIO(code)
+                    for chunk in iter(lambda: fcode.read(32), b''):
```

Comment:
chunking like this is nicer to humans visually, but makes it harder to copy/paste or parse from the file. i feel like we should just simply dump it